util/parsenum.h

```diff
@@ -53,7 +61,7 @@ _Pragma("clang diagnostic pop")
 		(((*(x)) = -1) > 0) ?					\
 			((*(x)) = parsenum_unsigned((s), 0, (*(x)),	\
 			    (*(x)))) :					\
-			(assert(0 && "PARSENUM applied to signed integer without specified bounds"), 1),	\
+			(ASSERT_MSG(0, "PARSENUM applied to signed integer without specified bounds"), 1),	\
```

Comment:
You checked that `assert(("foo", 0))` doesn't work?  (Note double parentheses, since this is all part of the first (and only) argument to `assert`.)

Reply:
Yes, that ran into "expression result unused", related to the string.